crates/wasmtime/src/store.rs

```diff
@@ -248,19 +248,23 @@ impl Store {
         &self.inner.signatures
     }
 
-    pub(crate) fn lookup_shared_signature<'a>(
-        &'a self,
-        types: &'a TypeTables,
-    ) -> impl Fn(wasm::SignatureIndex) -> VMSharedSignatureIndex + 'a {
-        move |index| {
-            self.signatures()
-                .borrow()
-                .lookup(&types.wasm_signatures[index])
-                .expect("signature not previously registered")
+    pub(crate) fn register_module(&self, module: &Module) {
+        // With a module being instantiated into this `Store` we need to
+        // preserve its jit-code. References to this module's code and
+        // trampolines are not owning-references so it's our responsibility to
+        // keep it all alive within the `Store`.
+        //
+        // If this module is already present in the store then we skip all
+        // further registration steps.
+        let first = self
+            .inner
+            .modules
+            .borrow_mut()
+            .insert(ArcModuleCode(module.compiled_module().code().clone()));
+        if !first {
+            return;
```

Comment:
Ah okay this is where all that stuff about whether we are registering for the first time or not is handled. (FWIW, this would have been a little easier on me, as a reviewer, if it were in a separate commit)

Reply:
Oops my bad, sorry about that! I forgot to call this out in the description as well...